dawn/python/dawn/sir.py

```diff
@@ -547,9 +547,33 @@ def make_field_access_expr(name: str, offset: List[int] = [0, 0, 0],
     :param argument_offset: Offset to the directional and offset arguments.
     :param negate_offset:   Negate the offset in the end.
     """
+    assert offset is None or isinstance(offset, list)
+
     expr = FieldAccessExpr()
     expr.name = name
-    expr.offset.extend(offset)
+    if offset is None:
+        expr.zero_offset.SetInParent()
+
+    elif len(offset) == 3:
+        assert all(isinstance(x, int) for x in offset)
+
+        expr.cartesian_offset.i_offset = offset[0]
+        expr.cartesian_offset.j_offset = offset[1]
+
+        expr.vertical_offset = offset[2]
+
+    elif len(offset) == 2:
```

Comment:
quite hacky, but I don't care at the moment ;)

Reply:
yes, I remember having seen code like this. I don't know what is the best way to do that, but I also didn't care that much right now :-)